#!/usr/bin/env python
PACKAGE = "ahl_robot_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp", double_t, 0, "Gain for position", 1.0, 0.0, 100.0)
gen.add("kv", double_t, 0, "Gain for velocity", 0.1, 0.0, 100.0)
gen.add("kv_damp", double_t, 0, "Gain for damping", 0.1, 0.0, 100.0)
gen.add("kp_limit", double_t, 0, "Position gain for joint limit", 1.0, 0.0, 100.0)
gen.add("kv_limit", double_t, 0, "Velocity gain for joint limit", 1.0, 0.0, 100.0)
gen.add("gx", double_t, 0, "Gravity acceleration along x", 0.0, -100.0, 100.0)
gen.add("gy", double_t, 0, "Gravity acceleration along y", 0.0, -100.0, 100.0)
gen.add("gz", double_t, 0, "Gravity acceleration along z", -0.980, -100.0, 100.0)

exit(gen.generate(PACKAGE, "ahl_robot_controller", "Param"))
